core/src/main/java/com/yahoo/ycsb/DBWrapper.java

```diff
@@ -26,157 +28,196 @@
-		_db.cleanup();
-        long en=System.nanoTime();
-        measure("CLEANUP",ist, st, en);
-	}
-
-	/**
-	 * Read a record from the database. Each field/value pair from the result will be stored in a HashMap.
-	 *
-	 * @param table The name of the table
-	 * @param key The record key of the record to read.
-	 * @param fields The list of fields to read, or null for all of them
-	 * @param result A HashMap of field/value pairs for the result
-	 * @return The result of the operation.
-	 */
-	public Status read(String table, String key, Set<String> fields, HashMap<String,ByteIterator> result)
-	{
-	    long ist=_measurements.getIntendedtartTimeNs();
-	    long st = System.nanoTime();
-	    Status res=_db.read(table,key,fields,result);
-		long en=System.nanoTime();
-		measure("READ",ist, st, en);
-	    _measurements.reportStatus("READ",res);
-		return res;
-	}
-
-	/**
-	 * Perform a range scan for a set of records in the database. Each field/value pair from the result will be stored in a HashMap.
-	 *
-	 * @param table The name of the table
-	 * @param startkey The record key of the first record to read.
-	 * @param recordcount The number of records to read
-	 * @param fields The list of fields to read, or null for all of them
-	 * @param result A Vector of HashMaps, where each HashMap is a set field/value pairs for one record
-	 * @return The result of the operation.
-	 */
-	public Status scan(String table, String startkey, int recordcount, Set<String> fields, Vector<HashMap<String,ByteIterator>> result)
-	{
-	    long ist=_measurements.getIntendedtartTimeNs();
-	    long st = System.nanoTime();
-	    Status res=_db.scan(table,startkey,recordcount,fields,result);
-		long en=System.nanoTime();
-		measure("SCAN",ist, st, en);
-	    _measurements.reportStatus("SCAN",res);
-		return res;
-	}
-
-    private void measure(String op, long intendedStartTimeNanos, long startTimeNanos, long endTimeNanos) {
-        _measurements.measure(op, (int)((endTimeNanos-startTimeNanos)/1000));
-	    _measurements.measureIntended(op, (int)((endTimeNanos-intendedStartTimeNanos)/1000));
+  DB _db;
+  Measurements _measurements;
+
+  boolean reportLatencyForEachError = false;
+  HashSet<String> latencyTrackedErrors = new HashSet<String>();
+
+  public DBWrapper(DB db)
+  {
+    _db=db;
+    _measurements=Measurements.getMeasurements();
+  }
+
+  /**
+   * Set the properties for this DB.
+   */
+  public void setProperties(Properties p)
+  {
+    _db.setProperties(p);
+  }
+
+  /**
+   * Get the set of properties for this DB.
+   */
+  public Properties getProperties()
+  {
+    return _db.getProperties();
+  }
+
+  /**
+   * Initialize any state for this DB.
+   * Called once per DB instance; there is one DB instance per client thread.
+   */
+  public void init() throws DBException
+  {
+    _db.init();
+
+    this.reportLatencyForEachError = Boolean.parseBoolean(getProperties().
+        getProperty("reportlatencyforeacherror", "false"));
```

Comment:
Can you pull the property string and default values into variables? You can see this convention in CoreWorkload: https://github.com/brianfrankcooper/YCSB/blob/master/core/src/main/java/com/yahoo/ycsb/workloads/CoreWorkload.java#L110-L117